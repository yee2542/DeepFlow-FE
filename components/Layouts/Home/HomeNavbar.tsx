/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { MenuOutlined } from '@ant-design/icons';
import { Button, Drawer, Menu } from 'antd';
import APP_NAVBAR_CONSTANT from 'Models/menu/app.navbar.constant';
import React, { useState } from 'react';
import pathJoin from 'Utils/path-join';
import style from './styles.module.css';

const HomeNavbar: React.FC = () => {
  const [visible, setVisible] = useState(false);

  const showDrawer = () => {
    setVisible(true);
  };

  const onClose = () => {
    setVisible(false);
  };

  return (
    <nav className={style.navbar}>
      <Button
        className={style.sideMenu}
        type="primary"
        icon={<MenuOutlined />}
        onClick={showDrawer}
      />

      <Drawer
        title="Web name"
        placement="left"
        closable={false}
        onClose={onClose}
        visible={visible}>
        {APP_NAVBAR_CONSTANT.map(({ key, label, link }) => {
          return (
            <li key={key}>
              <a href={pathJoin(link)}>{label}</a>
            </li>
          );
        })}
      </Drawer>
      <Menu className={style.menu} mode="horizontal" defaultSelectedKeys={['2']}>
        {APP_NAVBAR_CONSTANT.map(({ key, label, link }) => {
          return (
            <Menu.Item key={key}>
              <a href={pathJoin(link)}>{label}</a>
            </Menu.Item>
          );
        })}
      </Menu>
    </nav>
  );
};

export default HomeNavbar;
