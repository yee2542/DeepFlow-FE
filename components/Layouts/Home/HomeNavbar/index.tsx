/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { MenuOutlined } from '@ant-design/icons';
import {
  BtnSideMenu,
  Logo,
  MenuDiv,
  Nav,
} from 'Components/Layouts/Home/HomeNavbar/styled';
import RightNav from 'Components/Layouts/Home/HomeNavbar/RightNavbar';
import SideMenu from 'Components/Layouts/Home/HomeNavbar/SideMenu';
import React, { useState } from 'react';

interface HomeNavbarProps {
  visible?: boolean;
}

const HomeNavbar: React.FC<HomeNavbarProps> = (props) => {
  const [visible, setVisible] = useState(props.visible || false);

  const showDrawer = () => {
    setVisible(true);
  };

  const onClose = () => {
    setVisible(false);
  };

  return (
    <div>
      <Nav>
        <Logo>FluentSearch</Logo>
        <RightNav />

        <BtnSideMenu onClick={showDrawer}>
          <MenuDiv>
            <MenuOutlined />
          </MenuDiv>
        </BtnSideMenu>
      </Nav>
      <SideMenu onClose={onClose} visible={visible} />
    </div>
  );
};

export default HomeNavbar;
