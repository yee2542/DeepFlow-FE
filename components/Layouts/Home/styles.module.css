.navbar {
  background-color: white;
  padding-top: 1.5rem;
}

.menu {
  text-align: right;
  padding-left: 2rem;
  padding-right: 2rem;
}
