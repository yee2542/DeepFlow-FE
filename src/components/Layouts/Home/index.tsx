import { Layout } from 'antd';
import React from 'react';

const HomeLayout: React.FC = (props) => {
  const { children } = props;
  const { Content, Footer } = Layout;
  return (
    <div>
      <Content>
        <div>{children}</div>
      </Content>
      <Footer>Footer</Footer>
    </div>
  );
};

export default HomeLayout;
