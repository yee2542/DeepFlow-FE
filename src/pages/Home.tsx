import React from 'react';

const Home: React.FC = () => {
    return (
        <div>
            <div>Home Page</div>
        </div>
    );
};

export default Home;
