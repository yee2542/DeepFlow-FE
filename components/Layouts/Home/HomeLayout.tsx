import { Layout } from 'antd';
import React from 'react';
import HomeNavbar from './HomeNavbar';

const HomeLayout: React.FC = (props) => {
  const { children } = props;
  const { Content, Footer } = Layout;
  return (
    <div>
      <HomeNavbar />
      <Content>
        <div>{children}</div>
      </Content>
      <Footer>Footer</Footer>
    </div>
  );
};

export default HomeLayout;
