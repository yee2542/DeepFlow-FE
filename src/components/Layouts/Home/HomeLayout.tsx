import React from 'react';
import HomeNavbar from './HomeNavbar';

const HomeLayout: React.FC = (props) => {
  const { children } = props;
  return (
    <div>
      <HomeNavbar />
      <div>{children}</div>
    </div>
  );
};

export default HomeLayout;
