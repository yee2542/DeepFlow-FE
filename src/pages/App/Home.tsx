import HomeLayout from 'Components/Layouts/Home/HomeLayout';
import React from 'react';

const Home: React.FC = () => {
  return <HomeLayout>test</HomeLayout>;
};

export default Home;
