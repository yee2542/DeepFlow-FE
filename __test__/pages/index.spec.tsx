import HomeLayout from 'Components/Layouts/Home/HomeLayout';
import { mount, shallow } from 'enzyme';
import Home from 'Pages/index';
import React from 'react';

describe('Pages/Home test', () => {
  it('Render correcly', () => {
    const wrap = <Home />;
    expect(wrap).toMatchSnapshot();
  });

  it('Home should be exisiting', () => {
    const wrap = shallow(<Home />);
    expect(wrap.exists()).toBe(true);
  });

  it('Should contain Home Layout', () => {
    const wrap = mount(<Home />);
    expect(wrap.find(<HomeLayout />).exists()).toBe(true);
  });
});
