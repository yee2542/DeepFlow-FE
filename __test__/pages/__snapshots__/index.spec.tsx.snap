// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Home test Render correcly 1`] = `<Home />`;
